replicator/src/main/scala/com/evolutiongaming/kafka/journal/replicator/Batch.scala

```diff
@@ -5,6 +5,16 @@ import cats.syntax.all.*
 import com.evolutiongaming.kafka.journal.*
 import com.evolutiongaming.skafka.Offset
 
+/**
+ * Receives list of records from [[ReplicateRecords]], groups and optimizes similar or sequential actions, like:
+ *  - two or more `append` or `delete` actions are merged into one
+ *  - optimizes list of records to minimize load on Cassandra, like:
+ *    - aggregate effective actions by processing them from the youngest record down to oldest
+ *    - ignore/drop all actions, which are before last `Purge`
+ *    - ignore all `Mark` actions
+ *    - if `append`(s) are followed by `delete` all `append`(s), except last, are dropped
+ *    - at the end, apply all aggregated batches following order: `purge`, `append`s, `delete`
```

Comment:
please describe in the scaladoc the assumptions you make regarding input data with which this algorithm valid (ie it won't work correctly if deletes are made into future)